src/views/common/TimeScrollView/TimeScrollViewSpans.ts

```diff
@@ -0,0 +1,43 @@
+import { Matrix } from 'mathjs';
```

Comment:
This logic had been embedded within the TimeScrollView itself, but it'll be easier to follow if it's by itself with clearly-named API functions. This was mostly a copy-unchanged situation.